### DISCUSSION TITLE
Handling of Renewal of CA certificate

### DISCUSSION THREAD

**Discussion by Sai-Charan-Madhvaraj** [strimzi/strimzi-kafka-operator#10273] [General]
Hi Team, 

As per Strimzi, we see when we set "generateCertificateAuthority=true", the Cluster CA gets renewed with existing Key. and other option is to renew the private key as well.

<img width="876" alt="image" src="https://github.com/strimzi/strimzi-kafka-operator/assets/95335222/ed182afd-368a-4a9f-ac0c-4d5b2256a25b">

 Do we have option to not renew the Cluster CA and just renew the respective certs (brokers, zk, etc) which are signed by this particular signing authority. Reason for this clients can use the same truststore (generally root and intermediate signing authorities have longer duration than actual certs ) and have a separate renewal period for Cluster CA and individual component certs.

**Comment by scholzj** [comment]
I don't think such things makes much sense. The server certificates have the same validity as the Strimzi-based CA and they would anyway not work when the CA expires. So it does not make sense to renew them separately. So why would you want to do that?

**Reply by Sai-Charan-Madhvaraj** [reply]
The only use case would be clients do not need to keep modifying the truststore when ever server certificates renew. 

But at the same time we can have server certs renewed at regular intervals and dont need longer validity period.

In this case we need to have condition that cluster ca validity should be longer than server cert.

**Reply by scholzj** [reply]
But you would still need to renew the CA sooner or later.

**Reply by Sai-Charan-Madhvaraj** [reply]
Yes.. For example, if we have validity of CA of 5 years and server certificate of 1 year. Until 5 years clients dont need to keep changing every year and whenever CA renews we have to renew server cert as well and for clients only Cluster CA matters but not server certs.

**Reply by scholzj** [reply]
I don't think you have that with the Strimzi-based CA as the validity is the same for the CA as for the server certificates.

**Reply by Sai-Charan-Madhvaraj** [reply]
Okay. thanks for confirmation.

on another note , Do you think is it something we can try to implement ?

**Reply by scholzj** [reply]
Sorry, but implement what? As I said, the certificates are renewed together. So you cannot really renew just one of them because the other would expire.
